emu_mps/hamiltonian.py

```diff
@@ -3,309 +3,351 @@
 to the Hamiltonian of a neutral atoms quantum processor.
 """
 
+from abc import abstractmethod, ABC
+from typing import Iterator
+
 from emu_base import HamiltonianType
 import torch
-
 from emu_mps.mpo import MPO
 
 dtype = torch.complex128  # always complex128
-iden_op = torch.eye(2, 2, dtype=dtype)  # dtype is always complex128
-n_op = torch.tensor([[0.0, 0.0], [0.0, 1.0]], dtype=dtype)
-creation_op = torch.tensor([[0.0, 1.0], [0.0, 0.0]], dtype=dtype)
-sx = torch.tensor([[0.0, 0.5], [0.5, 0.0]], dtype=dtype)
-sy = torch.tensor([[0.0, -0.5j], [0.5j, 0.0]], dtype=dtype)
-pu = torch.tensor([[0.0, 0.0], [0.0, 1.0]], dtype=dtype)
-
-
-def truncate_factor(
-    factor: torch.Tensor,
-    left_interactions: torch.Tensor,
-    right_interactions: torch.Tensor,
-    hamiltonian_type: HamiltonianType,
-) -> torch.Tensor:
-    if hamiltonian_type == HamiltonianType.XY:
-        left_interactions = torch.stack(
-            (left_interactions, left_interactions), dim=-1
-        ).reshape(-1)
-        right_interactions = torch.stack(
-            (right_interactions, right_interactions), dim=-1
-        ).reshape(-1)
-    padding = torch.tensor([True] * 2)
-    trunc = factor[torch.cat((padding, left_interactions))]
-    return trunc[:, :, :, torch.cat((padding, right_interactions))]
-
-
-def _first_factor_rydberg(interaction: bool) -> torch.Tensor:
-    """
-    Creates the first Ising Hamiltonian factor.
-    """
-    fac = torch.zeros(1, 2, 2, 3 if interaction else 2, dtype=dtype)
-    fac[0, :, :, 1] = iden_op
-    if interaction:
-        fac[0, :, :, 2] = n_op  # number operator
 
-    return fac
 
+class Operators:
+    id = torch.eye(2, 2, dtype=dtype)
```

Comment:
`id = torch.eye(2, dtype=dtype)` does the same

Reply:
done